dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -680,6 +609,165 @@ def test_json_decode_after_update(self):
+            }
+
+            expected_diff = {
+                "statistics": {
+                    "avg_predictions": {"a": "unchanged", "b": -0.7, "c": 0.7},
+                    "label_representation": {"a": -0.84, "b": "unchanged", "c": 0.84},
+                },
+                "data_label": [["a"], [], ["b"]],
+            }
+            self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+        # Test disabling one datalabeler profile for compiler diff
+        options = StructuredOptions()
+        options.data_labeler.is_enabled = False
+        compiler1 = col_pro_compilers.ColumnDataLabelerCompiler(data, options)
+        expected_diff = {}
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+        # Test disabling both datalabeler profiles for compiler diff
+        compiler2 = col_pro_compilers.ColumnDataLabelerCompiler(data, options)
+        expected_diff = {}
+        self.assertDictEqual(expected_diff, compiler1.diff(compiler2))
+
+    def test_json_encode(self):
+
+        compiler = col_pro_compilers.ColumnDataLabelerCompiler()
+
+        serialized = json.dumps(compiler, cls=ProfileEncoder)
+        expected = json.dumps(
+            {
+                "class": "ColumnDataLabelerCompiler",
+                "data": {
+                    "name": None,
+                    "_profiles": {},
+                },
+            }
+        )
+        self.assertEqual(expected, serialized)
+
+    def test_json_encode_after_update(self):
+
+        data = pd.Series(["-2", "-1", "1", "2"], name="test")
+        with test_utils.mock_timeit():
+            compiler = col_pro_compilers.ColumnDataLabelerCompiler(data)
+
+        with mock.patch.object(
+            compiler._profiles["order"], "__dict__", {"an": "order"}
+        ):
+            with mock.patch.object(
+                compiler._profiles["category"], "__dict__", {"this": "category"}
+            ):
+                serialized = json.dumps(compiler, cls=ProfileEncoder)
+
+        expected = json.dumps(
+            {
+                "class": "ColumnDataLabelerCompiler",
+                "data": {
+                    "name": "test",
+                    "_profiles": {
+                        "order": {"class": "OrderColumn", "data": {"an": "order"}},
+                        "category": {
+                            "class": "CategoricalColumn",
+                            "data": {"this": "category"},
+                        },
+                    },
+                },
+            }
+        )
+
+        self.assertEqual(expected, serialized)
+
+    def test_json_decode(self):
+        expected_compiler = col_pro_compilers.ColumnDataLabelerCompiler()
+        serialized = json.dumps(expected_compiler, cls=ProfileEncoder)
+
+        deserialized = load_compiler(json.loads(serialized))
+
+        test_utils.assert_profiles_equal(expected_compiler, deserialized)
+
+    def test_json_decode_after_update(self):
```

Comment:
some mocks in the DataLabelerColumn tests might be helpful for this wrt mocking out a data labeler and prediction capabilities to use "real" results

Reply:
yep